File: 111minimumDepthOfBinaryTree/minimumDepthOfBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if (Objects.isNull(root)) return 0;
        else
        {
            final var depth = Stream.of(root.left, root.right).map(this::minDepth).collect(Collectors.toList());
            final var min = Collections.min(depth);
            return 1 + (min != 0 ? min : Collections.max(depth));
        }
    }
}

File: 111minimumDepthOfBinaryTree/minimumDepthOfBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (!root) return 0
    else
    {
        const depth = [root.left, root.right].map(minDepth)
        return 1 + (Math.min(...depth) || Math.max(...depth))
    }
};
